tests/MongoDB.Driver.Tests/Specifications/retryable-writes/TestRunner.cs

```diff
@@ -75,34 +79,71 @@ private void VerifyFields(BsonDocument document, params string[] expectedNames)
             }
         }
 
-        private IMongoCollection<BsonDocument> InitializeCollection(BsonDocument definition)
+        private void InitializeCollection(BsonDocument definition)
         {
-            var connectionString = CoreTestConfiguration.ConnectionString.ToString();
-            var clientSettings = MongoClientSettings.FromUrl(new MongoUrl(connectionString));
-            clientSettings.RetryWrites = true;
-            var client = new MongoClient(clientSettings);
-            var database = client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
-            var collection = database.GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+            var client = DriverTestConfiguration.Client;
+            var database = client.GetDatabase(_databaseName);
+            var collection = database.GetCollection<BsonDocument>(_collectionName);
 
             database.DropCollection(collection.CollectionNamespace.CollectionName);
             collection.InsertMany(definition["data"].AsBsonArray.Cast<BsonDocument>());
-
-            return collection;
         }
 
-        private void RunTest(IMongoCollection<BsonDocument> collection, BsonDocument test, bool async)
+        private void RunTest(BsonDocument test, bool async)
         {
-            VerifyFields(test, "description", "failPoint", "operation", "outcome");
+            VerifyFields(test, "description", "clientOptions", "failPoint", "operation", "outcome");
+            var clientOptions = (BsonDocument)test.GetValue("clientOptions", null);
             var failPoint = (BsonDocument)test.GetValue("failPoint", null);
             var operation = test["operation"].AsBsonDocument;
             var outcome = test["outcome"].AsBsonDocument;
 
+            var client = CreateClient(clientOptions);
+            var database = client.GetDatabase(_databaseName);
+            var collection = database.GetCollection<BsonDocument>(_collectionName);
             var executableTest = CreateExecutableTest(operation);
-            using (ConfigureFailPoint(failPoint))
+
+            using (ConfigureFailPoint(client, failPoint))
             {
                 executableTest.Execute(collection, async);
-                executableTest.VerifyOutcome(collection, outcome);
             }
+
+            executableTest.VerifyOutcome(collection, outcome);
+        }
+
+        private IMongoClient CreateClient(BsonDocument clientOptions)
+        {
+            var clientSettings = ParseClientOptions(clientOptions);
+            return new MongoClient(clientSettings);
+        }
+
+        private MongoClientSettings ParseClientOptions(BsonDocument clientOptions)
+        {
+            var settings = DriverTestConfiguration.Client.Settings.Clone();
+
+            if (clientOptions == null)
+            {
+                settings.RetryWrites = true;
```

Comment:
Supposedly ALL the retryable writes tests have he new clientOptions document but some don't, hence this default.